pkg/cloud/externalconn/connectionpb/connection.proto

```diff
@@ -23,6 +28,8 @@ enum ConnectionType {
 }
 
 message ConnectionDetails {
+  ConnectionProvider provider = 1;
```

Comment:
Ah, here we are again :) This can be derived on the fly with a method in connection.go, right? It's 100% a function of `details` below? Why add a new field with redundant info?

Reply:
It is you're right, and I think this boils down to aesthetics. We need a way of identifying every registered external connection provider since we need to maintain a mapping from that to its constructor - https://github.com/cockroachdb/cockroach/pull/84931/files#diff-762ec8724bcbca595b9b4d66ddb1d22ac3befb4467a28da92228a50e7b0b4744R84. 

Now we could have a big switch in `connection.go` that takes in an empty `ConnectionDetails` and based on its type returns an enum value, and call that method prior to registering, but I don't know if that's more intuitive than what we have today? fwiw we have such a proto layout for `ExternalStorage` too, and it's proved easy to use when registering `limiter` cluster settings that should apply to only a particular provider type - https://github.com/cockroachdb/cockroach/blob/master/pkg/cloud/impl_registry.go#L58. That's not to say that we can't achieve this just with an additional `GetProvider(connectionDetails)`  call. Another unexpected benefit of sticking this in the proto is that `SELECT connection_name, connection_type, crdb_internal.pb_to_json('cockroach.cloud.externalconn.connectionpb.ConnectionDetails', connection_details) 
FROM system.external_connections;` actually prints out the provider too, and you don't have to remember what connection details maps to what provider. 

Is there a misuse of this field that you're concerned about? I'm happy to make the change, I just lack the intuition which is why I'm defaulting to existing ways.

Reply:
OK, I'm convinced! No specific gripe - as you say, I think it's just differing aesthetics, in which case I'm happy to defer to the author :)